File: VponDMPExampleObjC/VponDMPExampleObjC/ViewController.m
//
//  ViewController.m
//  VponDMPExampleObjC
//
//  Created by Yi-Hsiang, Chien on 2020/11/23.
//

#import "ViewController.h"
#import <VponDataAnalytics/VponDataAnalytics.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _locationManager = [[CLLocationManager alloc] init];
    [_locationManager requestAlwaysAuthorization];
    [_locationManager requestWhenInUseAuthorization];
}

- (IBAction) sendEvent:(id)sender {
    VDATracker *tracker = [[VDATracker alloc] init];
    VDABuilder *builder = [VDABuilder createEventWithEventName:@"login" extraData:@{@"key": @"value"}];
    [tracker send:builder];
}


@end

File: VponDMPExampleSwift/VponDMPExampleSwift/ViewController.swift
//
//  ViewController.swift
//  VponDMPExampleSwift
//
//  Created by Yi-Hsiang, Chien on 2020/11/23.
//

import UIKit
import VponDataAnalytics
import CoreLocation

class ViewController: UIViewController {

    let locationManager = CLLocationManager()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        locationManager.requestAlwaysAuthorization()
        locationManager.requestWhenInUseAuthorization()
    }
    
    @IBAction func sendEvent() {
        let tracker = VDATracker()
        let builder = VDABuilder.createEventWithEventName("login", extraData: ["key": "value"])
        tracker.send(builder)
    }
}

